streambed-logged/src/lib.rs

```diff
@@ -638,7 +738,12 @@ impl Decoder for StorableRecordDecoder {
                 src.advance(src.len() - remaining.len());
                 Ok(Some(record))
             }
-            Err(e) if e == postcard::Error::DeserializeUnexpectedEnd => Ok(None),
+            Err(e)
+                if e == postcard::Error::DeserializeUnexpectedEnd
+                    && src.len() <= self.max_record_size =>
+            {
```

Comment:
It is now only ok to ask for more bytes if we've not exceeded our max record size.